src/main/java/frc/robot/subsystems/ControlPanelSubsystem.java

```diff
@@ -1,56 +1,66 @@
 import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
 import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.I2C.Port;
-import edu.wpi.first.wpilibj.util.Color;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.common.ControlPanelColor;
 import frc.robot.common.ControlPanelColorSensor;
-import frc.robot.common.ControlPanelColorSensor.ControlPanelColor;
+import java.util.Arrays;
+import java.util.logging.Logger;
 
 public class ControlPanelSubsystem extends SubsystemBase {
 
-    private Logger logger = Logger.getLogger("ControlPanelSubsystem");
-    private DoubleSolenoid cpSolenoid = new DoubleSolenoid(CONTROL_PANEL_SOLENOID_FWD, CONTROL_PANEL_SOLENOID_REV);
-    private WPI_VictorSPX cpMotor = new WPI_VictorSPX(CONTROL_PANEL_MOTOR);
-    private ControlPanelColorSensor colorSensor = new ControlPanelColorSensor(Port.kOnboard);
-    private DriverStation ds = DriverStation.getInstance();
-    private ControlPanelColor turnToColor;
+  private Logger logger = Logger.getLogger("ControlPanelSubsystem");
+  private DoubleSolenoid cpSolenoid = new DoubleSolenoid(CONTROL_PANEL_SOLENOID_FWD, CONTROL_PANEL_SOLENOID_REV);
+  private WPI_VictorSPX cpMotor = new WPI_VictorSPX(CONTROL_PANEL_MOTOR);
+  private ControlPanelColorSensor colorSensor = new ControlPanelColorSensor(Port.kOnboard,
+      Arrays.asList(ControlPanelColor.getValidColors()));
+  private DriverStation ds = DriverStation.getInstance();
+  private ControlPanelColor turnToColor;
 
-    public void spin(double speed) {
-        cpMotor.set(ControlMode.PercentOutput, speed);
-    }
+  public void spin(double speed) {
+    cpMotor.set(ControlMode.PercentOutput, speed);
+  }
 
-    public void setSolenoid(boolean extend) {
-        if (extend) {
-            cpSolenoid.set(Value.kForward);
-        } else{
-            cpSolenoid.set(Value.kReverse);
-        }
+  public void setSolenoid(boolean extend) {
+    if (extend) {
+      cpSolenoid.set(Value.kForward);
+    } else {
+      cpSolenoid.set(Value.kReverse);
     }
+    //        cpSolenoid.set(extend ? Value.kForward : Value.kReverse);
+  }
 
-    public ControlPanelColor getControlPanelColor() {
-        return ControlPanelColor.getColorFromMatch(colorSensor.getMatchedColor());
-    }
+  public ControlPanelColor getDetectedColor() {
+    return ControlPanelColor.valueOf(colorSensor.getDetectedColor());
+  }
 
-    public ControlPanelColor getTurnToColor() {
-        return turnToColor;
-    }
+  public ControlPanelColor getTurnToColor() {
+    return turnToColor;
+  }
 
-    @Override
-    public void periodic() {
-        if (ds.getGameSpecificMessage().isEmpty()) return;
+  private ControlPanelColor calculateTurnToColor() {
+    ControlPanelColor dsColor = ControlPanelColor.valueOf(ds.getGameSpecificMessage());
+    int totalColors = ControlPanelColor.getValidColors().length;
+    return ControlPanelColor.getValidColors()[(dsColor.ordinal() + 2) % totalColors];
+  }
```

Comment:
The turnToColor is based off of how the mechanism is designed. Therefore calculating the color should happen in this class. If the mechanism changes, the turnToColor may change, and therefore this class should be the class to change.